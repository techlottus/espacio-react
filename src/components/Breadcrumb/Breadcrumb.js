import React, { useEffect } from "react";

export const Breadcrumb = React.memo(({ data, onItem, onBack }) => {
  const breadcrumbRef = React.createRef();

  useEffect(() => {
    breadcrumbRef.current.data = {
      textItems: data.textItems || [],
      textColor: data.textColor || "brand",
      // iconColor: data.iconColor || '',
      // itemFocusColor: data.itemFocusColor || '',
      icon: data.icon || "",
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    breadcrumbRef.current.addEventListener("onItem", onItem);
    breadcrumbRef.current.addEventListener("onBack", onBack);
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-breadcrumb ref={breadcrumbRef}></lottus-breadcrumb>
    </>
  );
});
