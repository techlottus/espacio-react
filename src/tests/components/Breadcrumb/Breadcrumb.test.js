import React from "react";
import { mount } from "enzyme";
import { Breadcrumb } from "../../../components/Breadcrumb/Breadcrumb";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState = {
  data: {
    textItems: null,
    textColor: null,
    icon: null
  },
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Breadcrumb data={initState.data} />
    </Provider>
  );
};

describe("Test <Breadcrumb />", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });
});
