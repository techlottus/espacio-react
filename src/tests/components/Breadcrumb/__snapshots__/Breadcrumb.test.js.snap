// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Breadcrumb /> should init component 1`] = `<lottus-breadcrumb />`;
